components/draw/draw.service.ts

```diff
@@ -375,6 +423,41 @@ export class HsDrawService {
         source: this.source,
         type: /** @type {GeometryType} */ this.type,
         style: changeStyle ? changeStyle() : undefined,
+        condition: (e) => {
+          if (e.originalEvent.buttons === 1) {
+            //left click
+            return true;
+          } else if (e.originalEvent.buttons === 2) {
+            //right click
+            const minPoints = this.HsConfig.preserveLastSketchPoint ? 1 : 0;
+            if (this.type == 'Polygon') {
+              const vertexCount = this.draw.sketchLineCoords_?.length;
+              if (vertexCount >= 4 - minPoints) {
+                setTimeout(() => {
```

Comment:
Try refactoring the duplicate 
```
 setTimeout(() => {
                  if (minPoints == 0) {
                    this.removeLastPoint();
                  }
                  this.draw.finishDrawing();
                }, 250);
``` 
into seperate function so it's more readable.

Reply:
29c25580f13c56e014bb0b95065009ba68ec8403